File: projetJeuMaven/src/main/java/fr/projetjeu/model/ObjetInventaire.java
package fr.projetjeu.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "objetInventaire")
public class ObjetInventaire {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "objinv_id")
	private Integer objInventaireId;

	@ManyToOne
	@JoinColumn(name = "objinv_obj_id", nullable = false)
	private Objet objet;

	@ManyToOne
	@JoinColumn(name = "objinv_inv_id", nullable = false)
	private Inventaire inventaire;

	@Column(name = "qte_inventaire", nullable = false)
	int quantiteInventaire;

	public Integer getObjInventaireId() {
		return objInventaireId;
	}

	public void setObjInventaireId(Integer objInventaireId) {
		this.objInventaireId = objInventaireId;
	}

	public Objet getObjet() {
		return objet;
	}

	public void setObjet(Objet objet) {
		this.objet = objet;
	}

	public Inventaire getInventaire() {
		return inventaire;
	}

	public void setInventaire(Inventaire inventaire) {
		this.inventaire = inventaire;
	}

	public int getQuantiteInventaire() {
		return quantiteInventaire;
	}

	public void setQuantiteInventaire(int quantiteInventaire) {
		this.quantiteInventaire = quantiteInventaire;
	}

}

File: ProjetAngular/src/app/pagesweb/model/objet-inventaire.ts
import { Inventaire } from './inventaire';
import { Objet } from './objet';

export class ObjetInventaire {
  public get objet(): Objet | undefined {
    return this._objet;
  }
  public set objet(value: Objet | undefined) {
    this._objet = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  public get quantiteInventaire(): number | undefined {
    return this._quantiteInventaire;
  }
  public set quantiteInventaire(value: number | undefined) {
    this._quantiteInventaire = value;
  }
  public get inventaire(): Inventaire | undefined {
    return this._inventaire;
  }
  public set inventaire(value: Inventaire | undefined) {
    this._inventaire = value;
  }

  constructor(
    private _id?: number,
    private _inventaire?: Inventaire,
    private _objet?: Objet,
    private _quantiteInventaire?: number
  ) {}
}
